broker/core/utils/src/commonFunctions.js

```diff
@@ -598,6 +598,10 @@ function demux(stream, options) {
     stream.on('readable', onreadable);
     stream.once('end', onend);
     stream.once('error', onerror);
+
+    setTimeout(function() {
+      stream.destroy();
+    }, 2000);
```

Comment:
If we are not destroying the stream, it gets stuck causing `502` Bad Gateway error. So, destroying it was a necessary step.

Reply:
@visargsoneji  can you pls add examples from dockerode which you referred for this as a comment. Might be helpful as future reference.